### PR TITLE
[stdlib] Correct documentation formatting in CompilerProtocols

### PR DESCRIPTION
**Pull request by Jager-yoo**
Hi. I corrected some doc formatting in the `stdlib > CompilerProtocols.swift` file.
Please check out the changes. Thanks!

cc. @amartini51

### CODE REVIEW

File: stdlib/public/core/CompilerProtocols.swift [hunk 1]
```diff
@@ -658,7 +658,7 @@ public protocol ExpressibleByArrayLiteral {
 ///
 ///     let countryCodes = ["BR": "Brazil", "GH": "Ghana",
 ///                         "JP": "Japan", "US": "United States"]
-///     // 'countryCodes' has type [String: String]
+///     // 'countryCodes' has type '[String: String]'
```
Comment (Jager-yoo) [inline stdlib/public/core/CompilerProtocols.swift:661]: In this file, types are enclosed in single quotes. To maintain consistency, I have made similar adjustments.
For example: 'integers' has type '[Int]', 'emptyArray' has type '[Bool]'.

**Review by amartini51 — APPROVED** [review]
This looks good — thank you!

Confirmed that this file uses single quotes (') around type names. Confirmed the output from `print()` matches this PR's change.

**Comment by amartini51** [comment]
@swift-ci Please smoke test and merge.

**Comment by Jager-yoo** [comment]
Thank you, @amartini51. It looks like we have a pending reviewer, which is standard-librarians.
Is that a bot or a person? I hope this gets merged soon! :)

**Review by Azoy — APPROVED** [review]

**Comment by amartini51** [comment]
@swift-ci Please smoke test Linux

@swift-ci Please smoke test macOS

**Comment by amartini51** [comment]
@swift-ci Please smoke test Linux

**Pull request merged**